docs/architecture/adr-108-e2e-abci++.md

```diff
@@ -0,0 +1,305 @@
+# ADR 108: E2E tests for CometBFT's behaviour in respect to ABCI++.
+
+## Changelog
+- 2023-08-08: Initial version (@nenadmilosevic95)
+
+
+## Context
+
+ABCI++ defines the interface between the application and CometBFT. A part of the specification is the ABCI++ grammar that describes the sequences of calls that the application can expect from CometBFT.
```

Comment:
Please add a link to the grammar's section in the spec

Reply:
Added.